warehouse/utils/project.py

```diff
@@ -189,3 +189,42 @@ def destroy_docs(project, request, flash=True):
         request.session.flash(
             f"Deleted docs for project {project.name!r}", queue="success"
         )
+
+
+def archive_project(project: Project, request) -> None:
+    if (
+        project.lifecycle_status is None
+        or project.lifecycle_status == LifecycleStatus.QuarantineExit
+    ):
+        project.lifecycle_status = LifecycleStatus.Archived
+        project.record_event(
+            tag=EventTag.Project.ProjectArchiveEnter,
+            request=request,
+            additional={
+                "submitted_by": request.user.username,
+            },
+        )
+        request.session.flash("Project archived", queue="success")
+    else:
+        request.session.flash(
+            f"Cannot archive project with status {project.lifecycle_status}",
+            queue="error",
+        )
```

Comment:
style: Consider early-exit inversion of the logic, like so (pseudocode):

```python
if not <conditions that must be met>:
    flash "cannot archive", error
    return

do the work
flash "done", success
```

thus communicate early on the exit clause, continue normally otherwise.
Removes one layer of indentation for "the work", and makes it simpler to create another if/else when necessary with fewer logical jumps to make.

Similar applies to `unarchive_project`